timemachine/cpp/CMakeLists.txt

```diff
@@ -9,19 +9,31 @@ project(timemachine LANGUAGES CXX CUDA)
 find_package(PythonInterp 3.10 REQUIRED)
 find_package(PythonLibs 3.10 REQUIRED)
 
-string(APPEND CMAKE_CXX_FLAGS "-Werror -Wall -Wextra")
+set(DEBUG_BUILD "$ENV{DEBUG_BUILD}")
+
+set(COMMON_COMPILER_FLAGS "-Wall -Wextra")
+if (NOT DEBUG_BUILD)
+  string(APPEND COMMON_COMPILER_FLAGS " -O3")
+else()
+  string(APPEND COMMON_COMPILER_FLAGS " -g")
+endif()
+
+
+string(APPEND CMAKE_CXX_FLAGS "${COMMON_COMPILER_FLAGS} -Werror")
+string(APPEND CMAKE_CUDA_FLAGS "${COMMON_COMPILER_FLAGS}")
 
-string(APPEND CMAKE_CUDA_FLAGS "-O3 -lineinfo")
 string(APPEND CMAKE_CUDA_FLAGS
+  " -lineinfo"  # Disabling lineinfo reduces .so file by ~40MB
```

Comment:
Clarification: this change results in generating line info, not disabling it, correct? (Comment seems a bit ambiguous). Should line info be enabled only for debug builds?

Reply:
We have always generated line info, which is useful in practice. But didn't realize it made the .so so much larger. Just added a comment regarding the fact that it is taking up a significant amount of space by enabling it.